core/domain/exp_domain.py

```diff
@@ -1396,6 +1393,13 @@ def update_interaction_fallbacks(self, fallbacks_list):
         self.interaction.fallbacks = [
             Fallback.from_dict(fallback_dict)
             for fallback_dict in fallbacks_list]
+        if self.interaction.fallbacks:
+            hint_list = []
+            for fallback in self.interaction.fallbacks:
+                if fallback.outcome.feedback:
```

Comment:
Do you mean check for fallback.outcome then check for fallback.outcome.feedback? I think all fallbacks have non-empty outcomes, but some of those outcomes have no feedback (empty list).